src/test/java/teammates/storage/sqlapi/FeedbackQuestionsDbTest.java

```diff
@@ -0,0 +1,117 @@
+
+        mockHibernateUtil.verify(() -> HibernateUtil.persist(feedbackQuestion), times(1));
+    }
+
+    @Test
+    public void testCreateFeedbackQuestion_questionAlreadyExists_throwsEntityAlreadyExistsException() {
+        FeedbackQuestion feedbackQuestion = getFeedbackQuestion();
+        UUID fqid = feedbackQuestion.getId();
+
+        mockHibernateUtil.when(() -> HibernateUtil.get(FeedbackQuestion.class, fqid)).thenReturn(feedbackQuestion);
+
+        EntityAlreadyExistsException eaee = assertThrows(EntityAlreadyExistsException.class,
+                () -> feedbackQuestionsDb.createFeedbackQuestion(feedbackQuestion));
+
+        assertEquals(String.format(ERROR_CREATE_ENTITY_ALREADY_EXISTS, feedbackQuestion.toString()), eaee.getMessage());
+        mockHibernateUtil.verify(() -> HibernateUtil.persist(feedbackQuestion), never());
+    }
+
+    @Test
+    public void testCreateFeedbackQuestion_invalidQuestion_throwsInvalidParametersException() {
+        FeedbackQuestion feedbackQuestion = getFeedbackQuestion();
+        feedbackQuestion.setGiverType(FeedbackParticipantType.NONE);
+
+        InvalidParametersException ipe = assertThrows(InvalidParametersException.class,
+                () -> feedbackQuestionsDb.createFeedbackQuestion(feedbackQuestion));
+
+        assertEquals(feedbackQuestion.getInvalidityInfo(), List.of(ipe.getMessage()));
+        mockHibernateUtil.verify(() -> HibernateUtil.persist(feedbackQuestion), never());
+    }
+
+    @Test
+    public void testGetFeedbackQuestion_success() {
+        FeedbackQuestion feedbackQuestion = getFeedbackQuestion();
+        UUID fqid = feedbackQuestion.getId();
+
+        mockHibernateUtil.when(() -> HibernateUtil.get(FeedbackQuestion.class, fqid)).thenReturn(feedbackQuestion);
+
+        FeedbackQuestion retrievedSession = feedbackQuestionsDb.getFeedbackQuestion(fqid);
+
+        mockHibernateUtil.verify(() -> HibernateUtil.get(FeedbackQuestion.class, fqid), times(1));
+        assertEquals(feedbackQuestion, retrievedSession);
+    }
+
+    @Test
+    public void testGetFeedbackQuestion_questionDoesNotExist_returnNull() {
+        UUID fqid = UUID.randomUUID();
+
+        mockHibernateUtil.when(() -> HibernateUtil.get(FeedbackQuestion.class, fqid)).thenReturn(null);
+
+        FeedbackQuestion retrievedSession = feedbackQuestionsDb.getFeedbackQuestion(fqid);
+
+        mockHibernateUtil.verify(() -> HibernateUtil.get(FeedbackQuestion.class, fqid), times(1));
+        assertNull(retrievedSession);
+    }
+
+    @Test
+    public void testDeleteFeedbackQuestion_success() {
+        FeedbackQuestion feedbackQuestion = getFeedbackQuestion();
+        UUID fqid = feedbackQuestion.getId();
+
+        mockHibernateUtil.when(() -> HibernateUtil.get(FeedbackQuestion.class, fqid)).thenReturn(feedbackQuestion);
+
+        feedbackQuestionsDb.deleteFeedbackQuestion(fqid);
+
+        mockHibernateUtil.verify(() -> HibernateUtil.remove(feedbackQuestion), times(1));
+    }
```

Comment:
let's also add a a test case that ensures that delete is not called if feedbackquestion is null